pkgs/pkgs-lib/formats.nix

```diff
@@ -274,6 +274,68 @@ rec {
 
   };
 
+  # Format for defining configuration of some PHP services, that use "include 'config.php';" approach.
+  php = {finalVariable ? null}: with lib; let
+    toPHP = value: with builtins;
+      if value == null then "null" else
+      if value == true then "true" else
+      if value == false then "false" else
+      if isInt value || isFloat value then toString value else
+      if isString value then string value else
+      if isAttrs value then attrs value else
+      if isList value then list value else
+      abort "formats.php: should never happen (value = ${value})";
+
+    escapePHP = escape [ "'" ];
+    string = value: "'${escapePHP value}'";
+
+    listContent = values: concatStringsSep ", " (map toPHP values);
+    list = values: "[" + (listContent values) + "]";
+
+    attrsContent = values: pipe values [
+      (mapAttrsToList (k: v: "${toPHP k} => ${toPHP v}"))
+      (concatStringsSep ", ")
+    ];
+    attrs = set:
+      if set ? _phpType then specialType set
+      else
+          "[" + attrsContent set + "]";
+
+    mixedArray = {list, set}: "[${listContent list}, ${attrsContent set}]";
+
+    specialType = {value, _phpType}: {
+      "mixed_array" = mixedArray value;
+      "raw" = value;
+    }.${_phpType};
+
+    type = with lib.types;
+      nullOr (oneOf [
+        bool
+        int
+        float
+        str
+        (attrsOf type)
+        (listOf type)
+      ])
+      // {
+        description = "PHP value";
+      };
+  in {
+
+    inherit type;
+
+    lib = {
+      mkMixedArray = list: assert list != []; set: {_phpType = "mixed_array"; value = { inherit list set;}; };
+      mkRaw = raw: {_phpType = "raw"; value = raw;};
+    };
+
+    generate = name: value: pkgs.writeTextFile {
+      name = "config.php";
+      text = if finalVariable == null then "<?php\nreturn ${toPHP value};\n" else "<?php\n${finalVariable} = ${toPHP value};\n";
```

Comment:
Could we add `declare(strict_types=1);` ?

Reply:
What's the win from that? 

> Strict typing applies to function calls made from within the file with strict typing enabled, not to the functions declared within that file. If a file without strict typing enabled makes a call to a function that was defined in a file with strict typing, the caller's preference (coercive typing) will be respected, and the value will be coerced.

This only applies for function calls, which is very rare case in configs.

Reply:
That's true too.

Essentially, I use `declare(strict_types=1);` by default and for promoting best practices when one open a PHP file. To me it should be by default.

Reply:
Is it supported only in PHP 8? I can't find mention of versions on PHP website.

Reply:
It's available since ... PHP 4: https://3v4l.org/IePpo#v4.3.11

Reply:
Done